container/containerpostgres/postgres.go

```diff
@@ -54,67 +56,75 @@ func (p *Container) DSN() string {
 	return p.dsn
 }
 
-func New(t *testing.T) *Container {
+func New(t *testing.T, port ...int) *Container {
 	t.Helper()
 
-	addr := os.Getenv("POSTGRES_HOST")
-	var postgresContainer testcontainers.Container
-
-	if addr == "" {
-		image := DefaultPostgresImage
-		if v := os.Getenv("TEST_IMAGE_POSTGRES"); v != "" {
-			image = v
-		}
-
-		req := testcontainers.ContainerRequest{
-			Image:        image,
-			ExposedPorts: []string{"5432/tcp"},
-			Env: map[string]string{
-				"POSTGRES_HOST_AUTH_METHOD": "trust",
-			},
-			WaitingFor: wait.ForLog("database system is ready to accept connections").WithOccurrence(2),
-		}
-		container, err := testcontainers.GenericContainer(t.Context(), testcontainers.GenericContainerRequest{
-			ContainerRequest: req,
-			Started:          true,
-		})
-
-		if container == nil {
-			t.Fatalf("could not create postgres container: %v", err)
-		}
-
-		port, err := container.MappedPort(t.Context(), "5432")
-		if err != nil {
-			t.Fatalf("could not get mapped port: %v", err)
-		}
-
-		host, err := container.Host(t.Context())
-		if err != nil {
-			t.Fatalf("could not get host: %v", err)
-		}
-
-		addr = net.JoinHostPort(host, port.Port())
-
-		postgresContainer = container
+	image := DefaultPostgresImage
+	if v := os.Getenv("TEST_IMAGE_POSTGRES"); v != "" {
+		image = v
 	}
 
-	dsn := fmt.Sprintf("postgres://postgres@%s/postgres", addr)
+	var (
+		postgresContainer *postgres.PostgresContainer
+		err               error
+	)
+
+	if port != nil && len(port) > 0 && port[0] > 0 {
+		// Create Postgres container
+		postgresContainer, err = postgres.Run(t.Context(),
+			image,
+			postgres.WithDatabase("testdb"),
+			postgres.WithUsername("postgres"),
+			postgres.WithPassword("postgres"),
+			testcontainers.WithWaitStrategy(
+				wait.ForLog("database system is ready to accept connections").
+					WithOccurrence(2).
+					WithStartupTimeout(time.Second*5)),
+			testcontainers.WithExposedPorts(strconv.Itoa(port[0])+"/tcp"),
+		)
+	} else {
+		// Create Postgres container
+		postgresContainer, err = postgres.Run(t.Context(),
+			image,
+			postgres.WithDatabase("testdb"),
+			postgres.WithUsername("postgres"),
+			postgres.WithPassword("postgres"),
+			testcontainers.WithWaitStrategy(
+				wait.ForLog("database system is ready to accept connections").
+					WithOccurrence(2).
+					WithStartupTimeout(time.Second*5)),
+		)
+	}
```

Comment:
dublicated codes with adding just an option, we can appact option and add default option for exposed port